lib/netlibc/src/netlibc.c

```diff
@@ -155,25 +155,25 @@ char *get_file_extension(char *file_path) {
  *
  ****/
 result_t read_file_to_string(const char *file_path) {
-  FILE *file = fopen(file_path, "r");
-  if (file == NULL) {
-    return ERR("Error opening file: %s : %s", strerror(errno), file_path);
-  }
-
-  fseek(file, 0, SEEK_END);
-  u32 file_size = (u32)ftell(file);
-  fseek(file, 0, SEEK_SET);
+  long input_file_len;
+  FILE *input_file;
+  char *file_str;
 
-  char *file_content = (char *)malloc(file_size + 1);
-
-  size_t bytes_read = fread(file_content, 1, file_size, file);
-  if (bytes_read != file_size) {
-    return ERR("Error reading file");
-  }
-
-  file_content[file_size] = '\0';
-  fclose(file);
-  return OK(file_content);
+  if (!(input_file = fopen(file_path, "r")))
+    return ERR("Error opening file: %s : %s", strerror(errno), file_path);
```

Comment:
This code style (using comma operator after return), while concise, makes the code harder to read and maintain. Would recommended using a more conventional structure. Same as the ones below.